src/mesh/serial_mesh.C

```diff
@@ -1248,10 +1248,9 @@ void SerialMesh::stitching_helper (SerialMesh* other_mesh,
             {
               const std::vector<boundary_id_type>& ids =
                 other_mesh->get_boundary_info().boundary_ids(other_elem->get_node(n));
-              if (!ids.empty())
-                {
-                  this->get_boundary_info().add_node(this_elem->get_node(n), ids);
-                }
+              this->get_boundary_info().add_node(this_elem->get_node(n),
+                                                 std::set<boundary_id_type>(ids.begin(),
+                                                                            ids.end()));
```

Comment:
This makes me wonder if we should deprecate the old interface after all.  It seems mean to make user code do the same vector-to-set conversion that we could happily do for them just so they can avoid getting yelled at by libmesh_deprecated().